docs/ai-coder/ai-bridge/mcp.md

```diff
@@ -0,0 +1,64 @@
+# MCP
+
+[Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a mechanism for connecting AI applications to external systems.
+
+AI Bridge can connect to MCP servers and inject tools automatically, enabling you to centrally manage the list of tools you wish to grant your users.
+
+> [!NOTE]
+> Only MCP servers which support OAuth2 Authorization are supported currently.
+>
+> [_Streamable HTTP_](https://modelcontextprotocol.io/specification/2025-06-18/basic/transports#streamable-http) is the only supported transport currently. In future releases we will support the (now deprecated) [_Server-Sent Events_](https://modelcontextprotocol.io/specification/2025-06-18/basic/transports#backwards-compatibility) transport.
+
+AI Bridge makes use of [External Auth](../../admin/external-auth/index.md) applications, as they define OAuth2 connections to upstream services. If your External Auth application hosts a remote MCP server, you can configure AI Bridge to connect to it, retrieve its tools and inject them into requests automatically - all while using each individual user's access token.
+
+For example, GitHub has a [remote MCP server](https://github.com/github/github-mcp-server?tab=readme-ov-file#remote-github-mcp-server) and we can use it as follows.
+
+```bash
+CODER_EXTERNAL_AUTH_0_TYPE=github
+CODER_EXTERNAL_AUTH_0_CLIENT_ID=...
+CODER_EXTERNAL_AUTH_0_CLIENT_SECRET=...
+# Tell AI Bridge where it can find this service's remote MCP server.
+CODER_EXTERNAL_AUTH_0_MCP_URL=https://api.githubcopilot.com/mcp/
+```
+
+See the diagram in [Implementation Details](../reference.md#implementation-details) for more information.
+
+You can also control which tools are injected by using an allow and/or a deny regular expression on the tool names:
+
+```env
+CODER_EXTERNAL_AUTH_0_MCP_TOOL_ALLOW_REGEX=(.+_gist.*)
+CODER_EXTERNAL_AUTH_0_MCP_TOOL_DENY_REGEX=(create_gist)
+```
+
+In the above example, all tools containing `_gist` in their name will be allowed, but `create_gist` is denied.
+
+The logic works as follows:
+
+- If neither the allow/deny patterns are defined, all tools will be injected.
+- The deny pattern takes precedence.
+- If only a deny pattern is defined, all tools are injected except those explicitly denied.
+
+In the above example, if you prompted your AI model with "list your available github tools by name", it would reply something like:
+
+> Certainly! Here are the GitHub-related tools that I have available:
+>
+> 1. `bmcp_github_update_gist`
+> 2. `bmcp_github_list_gists`
```

Comment:
I think we should make this a code block. The docs get rendered a bit weirdly with a block quote:

<img width="632" height="190" alt="image" src="https://github.com/user-attachments/assets/839ff7c7-0e45-4de2-838c-a48eec49be1b" />

Reply:
This is a bug in our docs engine.